ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsRange.scala

```diff
@@ -46,19 +48,30 @@ private[events] object EventsRange {
     * @param connection SQL connection
     * @return Event Sequential ID range.
     */
-  def readEventSeqIdRange(endInclusive: Offset)(
+  def readEventSeqIdRange(endInclusive: Offset, dbType: DbType)(
       connection: java.sql.Connection
   ): EventsRange[Long] = {
     if (endInclusive == Offset.beforeBegin) EmptyEventSeqIdRange
-    else EmptyEventSeqIdRange.copy(endInclusive = readEventSeqId(endInclusive)(connection))
+    else EmptyEventSeqIdRange.copy(endInclusive = readEventSeqId(endInclusive, dbType)(connection))
   }
 
-  private def readEventSeqId(offset: Offset)(connection: java.sql.Connection): Long = {
+  private def readEventSeqId(offset: Offset, dbType: DbType)(
+      connection: java.sql.Connection
+  ): Long = {
     import com.daml.platform.store.Conversions.OffsetToStatement
     // This query could be: "select max(event_sequential_id) from participant_events where event_offset <= ${range.endInclusive}"
     // however tests using PostgreSQL 12 with tens of millions of events have shown that the index
     // on `event_offset` is not used unless we _hint_ at it by specifying `order by event_offset`
-    SQL"select max(event_sequential_id) from participant_events where event_offset <= ${offset} group by event_offset order by event_offset desc limit 1"
+    val query = dbType match {
+      // 2 - Need to figure out order by event_offset alternative
+      // 3 - Need to get group by to work field even works
+      case Oracle =>
+        SQL"select nvl(max(event_sequential_id),0) from participant_events where (dbms_lob.compare(event_offset, ${offset}) IN (0, -1)) fetch next 1 rows only"
```

Comment:
LOB comparison shouldn't be needed anymore after the merge of #9402. Please rebase and remove the special case for oracle.

Reply:
LOB is gone

oracle still needs different limit clause and the additional null check as when max returns null oracle complains